src/rust-crypto/rust-crypto.ts

```diff
@@ -1232,10 +1232,6 @@ export class RustCrypto extends TypedEventEmitter<RustCryptoEvents, RustCryptoEv
                 config,
             );
         }
-
-        // start tracking devices for any users already known to be in this room.
-        const members = await room.getEncryptionTargetMembers();
```

Comment:
This is what was triggering the `loadMembersIfNeeded`. Was called during initial sync when the m.room.encryption was processed.
I moved that code in the RoomEncryptor, it's now only getting the known members from existing state and not loading if neeed